smlb/datasets/experimental/watercolor_pigments_c19/watercolor_pigments_c19.py

```diff
@@ -240,42 +240,62 @@ def __init__(self, encoding: str = "one-hot", **kwargs):
         Parameters:
             encoding: how to featurize the two primary pigments in the binary mixtures;
                 possible values:
-                'one-hot': each primary pigment is a column; for example, 0.02 mL of P3
-                    would be encoded as (0, 0, 0.02, 0, ..., 0)
+                'one-hot': the first 13 entries one-hot encode pigmentA and the next 13 entries
+                    one-hot encode pigmentB. For example, 0.01 mL of P3 as PigmentA and 0.02 mL
+                    of P1 as PigmentB would be encoded as 0.01 in the 3rd index and 0.02 in the
+                    13+1=14th index. Everything else would be zero.
+                'amounts': each primary pigment is a column. For example, 0.02 mL of P3
+                    and 0.04 mL of P4 would be encoded as (0, 0, 0.02, 0.04, 0, ..., 0).
         """
 
-        self._encoding = params.enumeration(encoding, {"one-hot"})
+        self._encoding = params.enumeration(encoding, {"one-hot", "amounts"})
 
         super().__init__(**kwargs)
 
-    def _apply_one_hot(self, data: WatercolorPigments2019Dataset, primary: dict) -> TabularData:
+    @staticmethod
+    def _apply_one_hot(data: WatercolorPigments2019Dataset) -> TabularData:
         """One-hot features."""
 
-        # todo: treat primary features
-
-        dim = 2 * 13
+        num_pigments = len(data.COLOR_NAMES[1:])
+        dim = 2 * num_pigments
         features = np.zeros((data.num_samples, dim))
 
         for i, e, y in zip(range(data.num_samples), data.samples(), data.labels()):
             features[i, 0 + e["indexA"] - 1] = e["concentrationA"]
-            features[i, 13 + e["indexB"] - 1] = e["concentrationB"]
+            features[i, num_pigments + e["indexB"] - 1] = e["concentrationB"]
 
         labels = np.asarray(data.labels())
 
         return TabularData(data=features, labels=labels)
 
+    @staticmethod
+    def _apply_amounts(data: WatercolorPigments2019Dataset) -> TabularData:
+        """Ingredient amounts features."""
+        num_pigments = len(data.COLOR_NAMES[1:])
+        features = np.zeros((data.num_samples, num_pigments))
+
+        for i, sample in zip(range(data.num_samples), data.samples()):
+            features[i, sample["indexA"] - 1] += sample["concentrationA"]
+            features[i, sample["indexB"] - 1] += sample["concentrationB"]
+
+        labels = np.asarray(data.labels())
+        return TabularData(data=features, labels=labels)
+
     def apply(self, data: TabularData) -> TabularData:
         """Featurize WatercolorPigments2019Dataset."""
 
         data = params.instance(data, TabularData)
 
-        # set up look-up table for primary pigment data
+        # set up look-up table for primary pigment data (currently unused)
+        # todo: create features with pigment amounts
```

Comment:
```suggestion
        # TODO: create features with pigment amounts
```
Capitalization is a standard that makes it easier to find TODO comments.
While I'm at it: it's also nice to add an email or similar identifier to keep track of who's supposed TODO it (or who has the best context). Reference: https://google.github.io/styleguide/cppguide.html#TODO_Comments